runtime/onert/backend/train/KernelGenerator.cc

```diff
@@ -291,14 +316,30 @@ void KernelGenerator::visit(const ir::train::operation::ElementwiseActivation &n
   auto output_tensor = _tensor_reg->getPortableTensor(output_index);
   auto input_tensor = _tensor_reg->getPortableTensor(input_index);
 
-  auto back_prop_input_tensor = getBackPropIn(node, input_index);
-  auto back_prop_output_tensor = getBackPropOut(output_index);
-
   auto fn = std::make_unique<ops::ElementwiseActivationLayer>();
 
-  fn->configure(input_tensor, output_tensor, back_prop_input_tensor, back_prop_output_tensor,
-                node.param().alpha, node.param().beta,
-                convertElementwiseActivationType(node.param().op_type));
+  auto convertToInferActivationType = [](const ir::operation::ElementwiseActivation::Type &type) {
```

Comment:
Shouldn't `alpha` also be considered?

```
  switch (op_type)
  {
    case ElementwiseActivationType::kReLU:
... 
        if ((alpha == std::numeric_limits<float>::infinity() || alpha == 6.0f) && beta == 0.f)
        {
          cpu::ops::ElementwiseActivationLayer::configure(
            input, output, alpha, beta, cpu::ops::ElementwiseActivationType::kReLU);
```

Reply:
It's already checked inside `ElementwiseActivationLayer::configureBackward` (https://github.com/Samsung/ONE/blob/7baeaa92f4ae0f681152bcb7747d950db31ffe15/runtime/onert/backend/train/ops/ElementwiseActivationLayer.cc#L58).
Do you prefer to have it also here?

Reply:
Ah, the alpha is not an issue. Why do you define the `convertToInferActivationType` function again? Can't use `convertElementwiseActivationType` function?

Reply:
Here we're converting `ir::operation::ElementwiseActivation::Type` to `cpu::ops::ElementwiseActivationType` while `convertElementwiseActivationType` returns `train::ops::ElementwiseActivationType`. In the previous version this conversion has been made impicitly (https://github.com/Samsung/ONE/blob/8dd929184d502317b17b75fad547ba87ec641281/runtime/onert/backend/train/ops/ElementwiseActivationLayer.cc#L53-L61). But in general I agree that it will be great to introduce some unification of enum types or at least store helpers function in a one place.

Reply:
I understand. The train Kernel generator selectively calls the cpu and train configuration functions according to the `op.isRequiredForBackward()` value. In order to support this, these changes should be implmenet in the KernelGenerator. So, the activation type has to be converted to the cpu and train types depending on the situation.